packages/grid/_modules_/grid/components/toolbar/DensitySelector.tsx

```diff
@@ -23,17 +23,17 @@ export function DensitySelector() {
   const DensityOptions: Array<DensityOption> = [
     {
       icon: <DensityCompactIcon />,
-      label: 'Compact',
+      label: apiRef!.current.getText('toolbarDensityCompact'),
       value: DensityTypes.Compact,
     },
     {
       icon: <DensityStandardIcon />,
-      label: 'Standard',
+      label: apiRef!.current.getText('toolbarDensityStandard'),
       value: DensityTypes.Standard,
     },
     {
       icon: <DensityComfortableIcon />,
-      label: 'Comfortable',
+      label: apiRef!.current.getText('toolbarDensityComfortable'),
```

Comment:
Not sure about the api method name. It seems a bit too generic?
How about we rename it to `getLocaleText`? Also we could rename the hook to `useLocale ` if it handles node and strings. We could also create 2 methods, one for node and one for text ie getLocaleText, `getLocaleNode` so it can be used only in specific edge cases. And we can avoid errors where the node is returned instead of a plain text.
We can also consider using `i18n` instead of `locale`

Reply:
> How about we rename it to getLocaleText?

Sounds great. 

> We could also create 2 methods,

Why would it be better with 2 methods? What about when we need to ask for a function, do we add a third method? The returned type is already correct so I don't see  how it would help avoid issues more than we currently have.

Reply:
I can rename it to `getI18n` but I would prefer to have only one method since actually both of them will do the same thing. The only difference will be that we how going to have 2 API methods to maintain and to choose which one to use based on the useless. It's much simpler to have 1 that only selects an entry from an object and returns it.

For `i18n` instead of 'locale' are you talking about the prop name? Like `i18nText` or just `i18n` instead of `localeText`?

Reply:
To avoid, having 2 method then the interface should have a determined type for each key

```
interface LocaleText {
 labelKey: string;
 ChildrenValue: String | React.Node
otherKey: (count:number):string;
}

type LocaleTextProp = Partial<LocaleText>;
```

Reply:
>For i18n instead of 'locale' are you talking about the prop name? Like i18nText or just i18n instead of localeText?

`Locale` or `i18n` we align the prop and the hook to the same name

Reply:
That should be the case, the hook can do additional logic and can use more that one prop. I see them as 2 separate things, you can have a more generic hook, like `useI18n` that handles the text provided by the user, the locale name etc.

Reply:
In what way does it handle the locale name?

We could consider 2 hooks...

Reply:
I have lost track of the discussion. What problem are we trying to solve?

Reply:
It's not a problem per say but rather - why do we need to match the name of the prop to the name of the hook. Only a few hooks follow this but they are only responsible for one particular feature. I can imagine that for part 2 of that ticket we will have to add additional functionality to the hook and maybe more props if need.

Reply:
Oh, well, I guess that `localeText` for the prop, `useLocaleText` for the hook, and `getLocaleText` for the getter could make it easier to understand that it's the same "concern".

Reply:
Ok then, I'll go ahead and rename it 😉